src/bm_sim/_assert.cpp

```diff
@@ -31,4 +32,16 @@ void _assert(const char* expr, const char* file, int line) {
   std::abort();
 }
 
+void abort_msg(const char* error_msg, const SourceInfo* srcInfo) {
```

Comment:
it would be better IMO to move this function to https://github.com/p4lang/behavioral-model/blob/master/src/bm_sim/core/primitives.cpp along with the definition of `assert` and `assume`. This file was never intended to have a dependency on the logger.